File: src/main/java/browser/EdgePageInteractor.java
package browser;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class EdgePageInteractor {
    private final WebDriver driver;

    public EdgePageInteractor(WebDriver driver) {
        this.driver = driver;
    }

    public void openUrl(String url) {
        driver.get(url);
    }

    public WebElement findElement(By by) {
        return driver.findElement(by);
    }
}

File: src/browser/EdgePageInteractor.py
class EdgePageInteractor:
    def __init__(self, driver):
        self.driver = driver

    def open_url(self, url: str):
        """Open a given URL in the browser."""
        self.driver.get(url)

    def find_element(self, by, value):
        """Find and return an element on the page."""
        return self.driver.find_element(by, value)
